File: results/java_codex/CONVERT_DECIMAL_FRACTION_BINARY_NUMBER.java
string decimalToBinary ( double num, int k_prec ) {
  string binary = "";
  int Integral = ( int ) num;
  double fractional = num - Integral;
  while ( Integral > 0 ) {
    int rem = Integral % 2;
    binary += ( ( char ) ( rem + '0' ) );
    Integral /= 2;
  }
  binary = reverse ( binary );
  binary += ( '.' );
  while ( k_prec -- > 0 ) {
    fractional *= 2;
    int fract_bit = ( int ) fractional;
    if ( fract_bit == 1 ) {
      fractional -= fract_bit;
      binary += ( char ) ( 1 + '0' );
    }
    else {
      binary += ( char ) ( 0 + '0' );
    }
  }
  return binary;
}

File: TransCoder_tokenized_test_set_functions/python/CONVERT_DECIMAL_FRACTION_BINARY_NUMBER.py
def decimalToBinary ( num, k_prec ) :
    binary = ""
    Integral = int ( num )
    fractional = num - Integral
    while ( Integral ) :
        rem = Integral % 2
        binary += str ( rem );
        
        Integral //=2
    binary=binary[::-1]
    binary+='.'
    while(k_prec):
        fractional*=2
        fract_bit=int(fractional)
        if(fract_bit==1):
            fractional-=fract_bit
            binary+='1'
        else:
            binary+='0'
        k_prec-=1
    returnbinary
